Limber/MIXMockServer.cpp

```diff
@@ -0,0 +1,219 @@
+                            itemPayload.setMIXChannel(true);
+                            rosterUpdatePayload->addItem(itemPayload);
+
+                            session->sendElement(IQ::createRequest(IQ::Set, iq->getFrom(), iq->getID(), rosterUpdatePayload));
+                        } else {
+                            SWIFT_LOG(debug) << "Channel " << *channelJID << "not hosted on domain." <<std::endl;
+                        }
+                    } else {
+                        session->sendElement(IQ::createError(iq->getFrom(), iq->getID(), ErrorPayload::BadRequest, ErrorPayload::Cancel));
+                    }
+                }
+            }
+
+            //If request comes to particular channel supported by service.
+            if (stanza->getTo().isValid() && mixChannelRegistry_->hasMIXChannel(stanza->getTo())) {
+                if (iq->getPayload<DiscoItems>()->getNode() == "mix") {
+                    SWIFT_LOG(debug) << "Query: Channel Nodes Supported" << std::endl;
+                    auto responsePayload = std::make_shared<DiscoItems>();
+                    responsePayload->addItem(DiscoItems::Item(MIX::ParticipantsNode, JID(iq->getTo())));
+                    responsePayload->addItem(DiscoItems::Item(MIX::MessagesNode, JID(iq->getTo())));
+                    responsePayload->addItem(DiscoItems::Item(MIX::PresenceNode, JID(iq->getTo())));
+                    responsePayload->addItem(DiscoItems::Item(MIX::JIDMapNode, JID(iq->getTo())));
+                    session->sendElement(IQ::createResult(iq->getFrom(), iq->getTo(), iq->getID(), responsePayload));
+                }
+                else {
+                    session->sendElement(IQ::createError(iq->getFrom(), iq->getID(), ErrorPayload::BadRequest, ErrorPayload::Cancel));
+                }
+            }
+        }
+
+        MIXJoin::ref createJoinResult(const std::unordered_set<std::string>& nodes) {
+            auto joinResultPayload = std::make_shared<MIXJoin>();
+            for (auto node : nodes) {
+                joinResultPayload->addSubscription(node);
+            }
+            joinResultPayload->setJID(JID("123456#coven@example.tg"));
+            return joinResultPayload;
+        }
+
+    private:
+        IDGenerator idGenerator_;
+        PlatformXMLParserFactory xmlParserFactory;
+        UserRegistry* userRegistry_;
+        SimpleMIXChannelRegistry* mixChannelRegistry_;
+        BoostIOServiceThread boostIOServiceThread_;
+        std::shared_ptr<BoostConnectionServer> serverFromClientConnectionServer_;
+        std::vector< std::shared_ptr<ServerFromClientSession> > serverFromClientSessions_;
+        FullPayloadParserFactoryCollection payloadParserFactories_;
+        FullPayloadSerializerCollection payloadSerializers_;
+};
+
+int main() {
+    //Log::setLogLevel(Log::Severity::debug);
+
+    SimpleEventLoop eventLoop;
+    SimpleUserRegistry userRegistry;
+
+    userRegistry.addUser(JID("some@example.tg"), "mixtest");
```

Comment:
I think I would prefer it if you used example.com, which is guaranteed to never mask existing domains, instead of assuming there will never be a .tg TLD.